eng/common/scripts/Helpers/Resource-Helpers.ps1

```diff
@@ -75,14 +75,24 @@ function Get-PurgeableResources {
 
   Write-Verbose "Retrieving deleted Key Vaults from subscription $subscriptionId"
 
-  # Get deleted Key Vaults for the current subscription.
-  $deletedKeyVaults = @(Get-AzKeyVault -InRemovedState `
-    | Add-Member -MemberType NoteProperty -Name AzsdkResourceType -Value 'Key Vault' -PassThru `
-    | Add-Member -MemberType AliasProperty -Name AzsdkName -Value VaultName -PassThru)
+  # TODO: Remove try/catch handler for Get-AzKeyVault - https://github.com/Azure/azure-sdk-tools/issues/5315
+  # This is a temporary workaround since Az module >= 9.2.0 uses a more recent API
+  # version than is supported in the dogfood cloud environment:
+  #
+  #   | The resource type 'deletedVaults' could not be found in the namespace 'Microsoft.KeyVault' for api version '2022-07-01'. The supported api-versions are
+  #   | '2016-10-01,2018-02-14-preview,2018-02-14,2019-09-01,2021-04-01-preview,2021-06-01-preview,2021-10-01,2021-11-01-preview'.
+  try {
+    # Get deleted Key Vaults for the current subscription.
+    $deletedKeyVaults = @(Get-AzKeyVault -InRemovedState `
```

Comment:
Is there any way we can query to see if the api version exists or not?

Reply:
We can but at that point we may as well pin the Az module version to an older version (which we should discuss tomorrow as an approach regardless), because knowing which api version will be queried under the hood requires knowing the Az.KeyVault module version we're using.